llvm/docs/SYCLFrontEndDeviceCompilation.rst

```diff
@@ -0,0 +1,135 @@
+=================================
+SYCL Front-end device compilation
+=================================
+
+.. contents::
+   :local:
+
+.. _sycl_frontend_device_compilation:
+SYCL front-end device compiler parses input source, outlines device part of the
+code, applies additional restrictions on the device code (e.g. no exceptions or
+virtual calls), generates LLVM IR bitcode for the device code only and
+"integration header" which provides information like kernel name, parameters
+order and data type for the runtime library. Multiple LLVM IR bitcodes (in case
+of multiple targets) are packaged into a single object by the
+clang-offload-packager.
+
+An example of front-end device compilation command is shown below:
+
+  .. code-block:: console
+
+    $ clang -cc1 -triple spirv64 -fsycl-is-device test.cpp -o test.bc
```

Comment:
Can device compilation be illustrated using the driver directly without using `-cc1`? Perhaps with `-fsycl-device-only`? This should also pass `-aux-triple` to demonstrate the correlation with a host compilation.